llms/googleai/googleai.go

```diff
@@ -415,16 +415,21 @@ func convertTools(tools []llms.Tool) ([]*genai.Tool, error) {
 		}
 
 		if required, ok := params["required"]; ok {
-			ri := required.([]interface{})
-			rs := make([]string, 0, len(ri))
-			for _, r := range ri {
-				rString, ok := r.(string)
-				if !ok {
-					return nil, fmt.Errorf("tool [%d]: expected string for required", i)
+			if rs, ok := required.([]string); ok {
```

Comment:
Here is a demo, and I want to solve this problem:

```go
package test

import (
	"encoding/json"
	"fmt"
	"testing"
)

type Tool struct {
	Param any `json:"param"`
}

func Test_type(t *testing.T) {
	toolsByte := []byte(`{"param": ["a", "b", "c"]}`)
	var tool Tool
	_ = json.Unmarshal(toolsByte, &tool)
	// it with panic: interface conversion: interface {} is []interface {}, not []string
	s := tool.Param.([]string)
	fmt.Println(s)
}
```

Reply:
The issue I'm encountering is that json.Unmarshal decodes JSON arrays into slices of interface{}, not slices of string. To handle this, I need to manually convert the elements of the slice to string. This approach might seem complex, but it's necessary to ensure type safety.

Reply:
Yes, schema.Required does need a []string. However, because the upper-level variable is of type any, after serialization and transmission, the []string becomes []interface{}. To ensure type safety, it is necessary to manually convert the []interface{} back to []string. This approach helps avoid type errors and ensures data consistency.

Reply:
<img width="796" alt="image" src="https://github.com/tmc/langchaingo/assets/37948697/d1420a8a-2a67-4929-9d32-31dc1f0612cc">

Reply:
Erh, I agree we should cover each case.

Reply:
```
SKIP: TestVertexShared/testToolsWithInterfaceRequired-vertex
```

The test cases can't run in the CI. I've already tested them locally. Please merge it as soon as possible—this issue was caused by me.